Positioning terminology tooltip to be always fully visible

MacDada opened the pull request:
Currently it does not always fit within the page:

![image](https://github.com/home-assistant/home-assistant.io/assets/136261/5d9e6a5e-ba6d-4e8e-bca5-2b0e507de37e)

#resolves https://github.com/home-assistant/home-assistant.io/issues/31341

## Type of change

- [x] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logos and icons in [Brands repository](https://github.com/home-assistant/brands).
- [ ] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

MacDada committed:
Positioning terminology tooltip to be always fully visible

#resolves https://github.com/home-assistant/home-assistant.io/issues/31341

ludeeus commented on source/javascripts/terminology_tooltip.js:
```diff
@@ -0,0 +1,44 @@
+'use strict';
+
+[...document.getElementsByClassName('terminology')].forEach($terminology => {
```
Lets not prefix with `$`

ludeeus replied:
there is a few other variables that also do this in this file

MacDada replied:
> Lets not prefix with `$`

sure.

just to explain myself: this is a convention kinda taken from jQuery -> vars containing DOM objects are prefixed with `$` -> so it is easy to differentiate them from "normal" objects/var.

like in this case: `$terminology` (the current DOM object) vs `terminology` (the current "term")

MacDada replied:
https://github.com/home-assistant/home-assistant.io/pull/31355/commits/9fc502028c3aac90751ed5a8f5f6042d9c976e5c

ludeeus commented on source/javascripts/terminology_tooltip.js:
```diff
@@ -0,0 +1,44 @@
+
+    const terminology = $terminology.innerText;
+
+    const debug = message => {
```
This do not need to be here.

MacDada replied:
I understand you want to me to get rid of `console.debug()`, not just `debug()` helper? ;)

ludeeus replied:
Yeah, both of them 👍

MacDada replied:
https://github.com/home-assistant/home-assistant.io/pull/31355/commits/aa23edcb8f62957c2bfa1c6b416415dedd81d254

ludeeus commented on source/javascripts/terminology_tooltip.js:
```diff
@@ -0,0 +1,44 @@
+        console.debug(`Terminology tooltip (${terminology}): ${message}`);
+    }
+
+    const $tooltip = $terminology.querySelector('.terminology-tooltip');
```
While unlikely, this can be undefined if a class of "terminology" is added to an element that is not using the term tag

MacDada replied:
Yep. It will be a NPE. It will throw in the console. It will not reposition the tooltip. That's good enough for me.

Do you want me to check for `$tooltip` existence and explicitly throw, instead of the implicit NPE?

ludeeus replied:
I think we can just return if the queryselector returns nothing for this

MacDada replied:
> I think we can just return if the queryselector returns nothing for this

That would silence a most-likely error on the programmer part -> someone forgetting to grep when attaching the class name in a new context.

I know, in theory, it could be possible for the two contexts to coexist with the same class name  -> but I'd strongly advise against that, as it could lead to unexpected results (like "their" new code messing with these tooltips).

ludeeus replied:
In a web application with a lot of moving parts I would agree.
But in a statically hosted documentation site that is as likely as "terminology" being added to some random element.

I have no strong feelings either way for this 👍

MacDada replied:
i've just stumbled upon name collisions MANY times, and i became pretty pedantic to keeping global names distinct and separated

> In a web application with a lot of moving parts I would agree.

haha, yeah - that's my background -> i'm probably overthinking this one ;)

MacDada committed:
terminology_tooltip: removed debugs

resolves https://github.com/home-assistant/home-assistant.io/pull/31355/commits/7a0183402fe72f0a3f205db6a3457a183a597791#r1485840114

MacDada committed:
terminology_tooltip: removed $–convention from var names

resolves https://github.com/home-assistant/home-assistant.io/pull/31355/commits/7a0183402fe72f0a3f205db6a3457a183a597791#r1485839944

MacDada committed:
terminology_tooltip: let -> const

ludeeus reviewed (approved):
Thanks 👍

ludeeus committed:
Merge branch 'current' into tooltip
